src/macros.jl

```diff
@@ -317,9 +317,18 @@ macro grad_from_chainrules(fcall)
     Meta.isexpr(fcall, :call) && length(fcall.args) >= 2 ||
         error("`@grad_from_chainrules` has to be applied to a function signature")
     f = esc(fcall.args[1])
-    xs = fcall.args[2:end]
+    xs = map(fcall.args[2:end]) do x
+        if x isa Expr && x.head == :(::)
+            if length(x.args) == 1 # ::T without var name
+                return :($(gensym())::$(esc(x.args[1])))
```

Comment:
Why is a variable name needed? Can't we just escape the type?

Reply:
it was sure removing the var nam errored but it now works, let's see if works on CI

Reply:
I see. Removing the gensym breaks this line
```julia
$f($(args_l...)) = ReverseDiff.track($(args_r...))
```
making it throw the following syntax error

ERROR: syntax: invalid "::" syntax around ~/.julia/dev/ReverseDiff/src/macros.jl:338
Stacktrace:
 [1] top-level scope

Reply:
I am not sure why this errors, seems like valid Julia syntax to me.

Reply:
Seems to me the problem is not the LHS but rather the RHS: For instance, with the gensymed variable removed, we get
```julia
julia> @macroexpand ReverseDiff.@grad_from_chainrules sin(x::Float64)
quote
    #= /home/david/.julia/dev/ReverseDiff/src/macros.jl:333 =#
    sin(var"#50#x"::Float64) = begin
            #= /home/david/.julia/dev/ReverseDiff/src/macros.jl:333 =#
            (ReverseDiff.ReverseDiff).track(sin, var"#50#x"::Float64)
        end
...

julia> @macroexpand ReverseDiff.@grad_from_chainrules sin(::Float64)
quote
    #= /home/david/.julia/dev/ReverseDiff/src/macros.jl:333 =#
    sin(::Float64) = begin
            #= /home/david/.julia/dev/ReverseDiff/src/macros.jl:333 =#
            (ReverseDiff.ReverseDiff).track(sin, ::Float64)
        end
...
```
Seems like a bug in `_make_fwd_args` that we include the types in the `track` calls but in any case it seems we need the gensymed variables.